### PR TITLE
created test for core/rest.py

### LINKED ISSUE #189: Create unittests for `switchmap/core/rest.py`
We will need unittests done for all methods, classes and functions found in this file.

### PR Acceptance Criteria

1. When complete this file must show **100%** coverage when merged into the code base.
1. [The current code coverage for the file can be found here](https://app.codecov.io/gh/PalisadoesFoundation/switchmap-ng/tree/develop/?displayType=list)
1. CodeRabbit must approve all your changes to the test file(s) and the file(s) under test
1. The PR will show a report for the code coverage for the file you have added. You can use that as a guide
1. No functionality must be removed to achieve this goal
1. All tests must be valid and truly test the code base
1. The tests must be placed in the equivalent folder under `tests/switchmap_` therefore:
    1. a file named `switchmap/x/y/z/file.py` must have a test file named `tests/switchmap_/x/y/z/test_file.py`
    1. file `tests/switchmap_/x/y/z/test_file.py` must have 755 permissions so it can be easily executable
    1. take guidance on the formatting of the file from other `tests/switchmap_/x/y/z/test_*.py` files or others along the path or in adjacent directories
    
### Other Criteria

1. If this file isn't referenced in the code base then remove it. We don't want tests run on unused code

### Potential internship candidates

Please read this if you are planning to apply for a Palisadoes Foundation internship 
- https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Dhiren-Mhatre**
## What kind of change does this PR introduce?  
Unit Testing

## Issue Number  
Fixes #189

## Summary  
This PR adds comprehensive unit tests for the `switchmap/core/rest.py` file, ensuring 100% code coverage. These tests validate all methods, classes, and functions in the file without altering the current functionality.

### Changes Made  
- **Test File Added:** `tests/switchmap_/core/test_rest.py`
- **Test Scenarios Covered:**
  - Valid REST API calls and responses.
  - Invalid input handling for API requests.
  - Edge cases and error scenarios.
- **Enhancements in `rest.py`:**
  - Improved exception handling for better test coverage.
  - Modularized certain code blocks to enhance readability and testability.
- **File Permissions:** Ensured 755 permissions for `test_rest.py` to align with project standards.
- **Consistency:** Followed formatting and structure from existing test files in the codebase.

### Snapshots  
image

### Test Coverage Details  
- Achieved 100% code coverage for `rest.py` as verified by the coverage report.
- Developed robust test cases for both success and failure scenarios, ensuring no untested edge cases remain.

## Does this PR introduce a breaking change?  
No - This PR focuses solely on adding unit tests and refining code without affecting existing functionality.

---

### CodeRabbit AI Review  
- [x] Reviewed and addressed all critical issues flagged by CodeRabbit AI.  
- [x] Implemented or justified all non-critical suggestions.  
- [x] Documented reasoning where suggestions were not implemented.

### Test Coverage  
- [x] Wrote tests for all new changes/features.  
- [x] Verified test coverage is 100%.  
- [x] All tests pass when run locally.  
- [x] Tests avoid external dependencies using mocking where needed.  
- [x] Both success and failure scenarios are tested.

## Have you read the [contributing guide](https://github.com/PalisadoesFoundation/switchmap-ng/blob/main/CONTRIBUTING.md)?  
Yes.

## Summary by CodeRabbit

- **Tests**
	- Added comprehensive test suite for the REST module.
	- Implemented multiple test classes to validate REST functionality.
	- Enhanced test coverage for error handling and edge cases.
	- Introduced mock configurations and simulated HTTP request scenarios.
	- Ensured robust validation of API behavior under various scenarios.
- **Bug Fixes**
	- Improved error handling in the post function to ensure consistent behavior by defining the result variable in all control paths.

### CODE REVIEW

**Review by palisadoes — CHANGES_REQUESTED** [review]
Please use the flask-testing package and not mocks for this.
- https://flask.palletsprojects.com/en/stable/testing/
- https://www.digitalocean.com/community/tutorials/unit-test-in-flask

We need to be absolutely certain that the code always works

**Comment by palisadoes** [comment]
Does the test URI exist in flask? Search their site to see how you can verify this

File: tests/switchmap_/core/test_rest.py [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+        """Test the _clean_url utility function directly."""
+        urls = [
+            "http://example.com//api//v1//data",
+            "https://example.com//api//v1//data",
+            "http:/example.com/api/v1/data",
+            "https:/example.com/api/v1/data",
+            "http://example.com////////",
+        ]
+        for url in urls:
+            _ = rest._clean_url(url)
```
Comment (palisadoes) [inline tests/switchmap_/core/test_rest.py:306]: What is this testing?

**Review by palisadoes — CHANGES_REQUESTED** [review]
See comment

File: tests/switchmap_/core/test_rest.py (hunk 1 above)
Comment (Dhiren-Mhatre) [reply tests/switchmap_/core/test_rest.py]: This test was initially created as part of test_rest.py when I was using mocks for URL validation. Since the mocks have been replaced or are no longer in use for this functionality
